parse/src/main/java/com/parse/ParseObjectStoreMigrator.java

```diff
@@ -0,0 +1,71 @@
+package com.parse;
+
+import com.parse.boltsinternal.Continuation;
+import com.parse.boltsinternal.Task;
+
+import java.util.Arrays;
+
+public class ParseObjectStoreMigrator<T extends ParseObject> implements ParseObjectStore<T> {
+
+    private ParseObjectStore<T> store;
+    private ParseObjectStore<T> legacy;
+
+    public ParseObjectStoreMigrator(ParseObjectStore<T> store, ParseObjectStore<T> legacy) {
+        this.store = store;
+        this.legacy = legacy;
+    }
+
+    private static <T extends ParseObject> Task<T> migrate(
```

Comment:
Is there a way to check whenever migrate is needed?

Reply:
If `from` is deleted and next time you call migrate the `onSuccessTask` what will happen? It will call again `onSuccessTask`? Is there chance `onFailedTask` or something similar to be called in such case?

Reply:
Thanks for pointing that out, that is actually a bug which is going to be calling `from.getAsync()` recursively.